supla-common/proto.h

```diff
@@ -499,24 +518,48 @@ extern char sproto_tag[SUPLA_TAG_SIZE];
 #define SUPLA_MFR_COMELIT 14
 #define SUPLA_MFR_POLIER 15
 
+// BIT map definition for TDS_SuplaRegisterDevice_*::Flags (32 bit)
 #define SUPLA_DEVICE_FLAG_CALCFG_ENTER_CFG_MODE 0x0010  // ver. >= 17
-#define SUPLA_DEVICE_FLAG_SLEEP_MODE_ENABLED 0x0020     // ver. >= 18
-
+#define SUPLA_DEVICE_FLAG_SLEEP_MODE_ENABLED    0x0020  // ver. >= 18
+#define SUPLA_DEVICE_FLAG_CALCFG_SET_TIME       0x0040  // ver. >= 20
+
+// BIT map definition for TDS_SuplaRegisterDevice_F::ConfigFields (64 bit)
+// type: TDeviceConfig_StatusLed
+#define SUPLA_DEVICE_CONFIG_FIELD_STATUS_LED            (1ULL << 0)  // v. >= 20
+// type: TDeviceConfig_ScreenBrightness
+#define SUPLA_DEVICE_CONFIG_FIELD_SCREEN_BRIGHTNESS     (1ULL << 1)  // v. >= 20
+// type: TDeviceConfig_ButtonVolume
+#define SUPLA_DEVICE_CONFIG_FIELD_BUTTON_VOLUME         (1ULL << 2)  // v. >= 20
+// type: TDeviceConfig_DisableLocalConfig
+#define SUPLA_DEVICE_CONFIG_FIELD_DISABLE_LOCAL_CONFIG  (1ULL << 3)  // v. >= 20
+// type: TDeviceConfig_TimezoneOffset
+#define SUPLA_DEVICE_CONFIG_FIELD_TIMEZONE_OFFSET       (1ULL << 4)  // v. >= 20
+// type: TDeviceConfig_AutomaticTimeSync
+#define SUPLA_DEVICE_CONFIG_FIELD_AUTOMATIC_TIME_SYNC   (1ULL << 5)  // v. >= 20
+// type: TDeviceConfig_ScreensaverDelay
+#define SUPLA_DEVICE_CONFIG_FIELD_SCREENSAVER_DELAY     (1ULL << 6)  // v. >= 20
+// type: TDeviceConfig_ScreensaverMode
+#define SUPLA_DEVICE_CONFIG_FIELD_SCREENSAVER_MODE      (1ULL << 7)  // v. >= 20
+
+// BIT map definition for TDS_SuplaDeviceChannel_C::Flags (32 bit)
 #define SUPLA_CHANNEL_FLAG_ZWAVE_BRIDGE 0x0001  // ver. >= 12
 #define SUPLA_CHANNEL_FLAG_IR_BRIDGE 0x0002     // ver. >= 12
 #define SUPLA_CHANNEL_FLAG_RF_BRIDGE 0x0004     // ver. >= 12
 // Free bit for future use: 0x0008
 #define SUPLA_CHANNEL_FLAG_CHART_TYPE_BAR 0x0010                  // ver. >= 12
+                                                                  // DEPRECATED
 #define SUPLA_CHANNEL_FLAG_CHART_DS_TYPE_DIFFERENTAL 0x0020       // ver. >= 12
+                                                                  // DEPRECATED
```

Comment:
This was to be used on a general purpose measurement channel.
If it really needs to be "Deprecated", it's worth renaming this macro by adding _Deprecated at the end.